src/GafferSceneUI/VisualiserTool.cpp

```diff
@@ -223,430 +417,1003 @@ class VisualiserGadget : public Gadget
+										data->writable() = v3fData->readable()[i];
+										vertexValue = data;
+									}
+									if( auto c3fData = runTimeCast<const Color3fVectorData>( vData.get() ) )
+									{
+										auto data = runTimeCast<Color3fData>( vertexValue );
+										if( !data )
+										{
+											data.reset( new Color3fData() );
+										}
+										data->writable() = c3fData->readable()[i];
+										vertexValue = data;
+									}
+								}
+
+								// Update cursor value
+								//
+								// NOTE : We defer drawing of the value currently under the cursor, so
+								//        draw the last value label if we replace the cursor value
+
+								const float distance2 = ( cursorRasterPos - rasterPos ).length2();
+								if( ( distance2 < cursorRadius2 ) && ( distance2 < minDistance2 ) )
+								{
+									std::swap( cursorVertexData, vertexValue );
+									std::swap( cursorVertexRasterPos, rasterPos );
+									minDistance2 = distance2;
+								}
+
+								// Draw value label
+
+								if( vertexValue )
+								{
+									const std::string text = stringFromData( vertexValue.get() );
+
+									drawStrokedText(
+										viewportGadget,
+										text,
+										m_tool->sizePlug()->getValue(),
+										V2f(
+											rasterPos.x - style->textBound( GafferUI::Style::LabelText, text ).size().x * 0.5f * scale.x,
+											rasterPos.y
+										),
+										style,
+										Style::State::NormalState
+									);
+								}
+							}
+						}
+					}
 
-						glTranslatef( -g_textShadowOffset * 2.f, 0.f, 0.f );
-						style->renderText( Style::LabelText, text, GafferUI::Style::State::NormalState, &g_textShadowColor );
+					// unmap storage buffer
 
-						glTranslatef( g_textShadowOffset, g_textShadowOffset, 0.f );
-						style->renderText( Style::LabelText, text, GafferUI::Style::State::NormalState, &g_textShadowColor );
+					glBindBuffer( GL_SHADER_STORAGE_BUFFER, m_vertexLabelStorageBuffer->buffer() );
+					glUnmapBuffer( GL_SHADER_STORAGE_BUFFER );
+					glBindBuffer( GL_SHADER_STORAGE_BUFFER, storageBinding );
+				}
 
-						glTranslatef( 0.f, -g_textShadowOffset * 2.f, 0.f );
-						style->renderText( Style::LabelText, text, GafferUI::Style::State::NormalState, &g_textShadowColor );
+				glBindBuffer( GL_SHADER_STORAGE_BUFFER, storageBinding );
+			}
 
-						/// Primary text
-						glTranslatef( 0.f, g_textShadowOffset, 0.f );
-						style->renderText( Style::LabelText, text );
+			// Draw cursor vertex
 
-						glPopMatrix();
-					}
-				}
+			if( cursorVertexData )
+			{
+				GafferUI::ViewportGadget::RasterScope raster( viewportGadget );
+
+				std::string const text = stringFromData( cursorVertexData.get() );
+
+				drawStrokedText(
+					viewportGadget,
+					text,
+					scale.x * 2.f,
+					V2f(
+						cursorVertexRasterPos.x - style->textBound( GafferUI::Style::LabelText, text ).size().x * scale.x,
+						cursorVertexRasterPos.y
+					),
+					style,
+					Style::State::HighlightedState
```

Comment:
I'm finding the blue text for the hovered vertex value to be less readable than the white. I think perhaps the size change is enough on its own.

Reply:
Yes, the white color with size difference seems good in a variety of situations : https://github.com/GafferHQ/gaffer/pull/6212/commits/674b5d7f98a3b04321e2c327b955174d655ca450